Add Image Captions on Thumbnail View

warrenhbrown opened the pull request:
This pull request adds support for displaying a caption under the images in the thumbnail view.  settings.json is extended with: displayImgCaption: when set to true, the fields (as configured in imageInfoFlds, see above) will render underneath preview views and the image detail view.

jyoung-uf committed:
Adjust formatting

jyoung-uf committed:
Conditionally render captions under images (grid and single)

jyoung-uf committed:
Fix formatting

jyoung-uf committed:
Update guide with image caption setting

jyoung-uf committed:
Fix typo

grantfitzsimmons commented on PortalApp/index.html:
```diff
@@ -81,6 +81,9 @@
     <script type="text/javascript" src="//maps.googleapis.com/maps/api/js?key=AIzaSyDYHJjsVAniNFUxSZnIO6pyVBxBqRt9gmw&sensor=false"></script>
     <!--script type="text/javascript" src="//maps.google.com/maps/api/js?sensor=false"></script-->
 
+    <!-- Google Tag Manager -->
```
It looks like the same changes are made on #76

https://github.com/specify/webportal-installer/blob/856d6391581ca6fe7eb6a33eeef142a6abebf4ae/PortalApp/index.html#L84-L85

melton-jason replied:
If this was intentional, it may be worth noting that the Pull Requests are now 'out of sync': there is an extra commit in https://github.com/specify/webportal-installer/pull/76 which is not present in this PR: https://github.com/specify/webportal-installer/pull/76/commits/db2e7f23c008051cb41a23146af46eb29b99c339

grantfitzsimmons reviewed (approved):
It looks like this PR includes the changes for #76 as well as the changes to add image captions in thumbnail views.

I really appreciate the way this was added (adding a new setting named `displayImgCaption` that can be toggled on or off) to make it simple and optional for users. Great work and thank you so much @warrenhbrown!

grantfitzsimmons commented:
@specify/dev-testing This code is live on their [Fish Web Portal](https://specifyportal.floridamuseum.ufl.edu/fishes/). When testing, we need to set up a new instance and verify that the `settings.json` can include a line that sets `displayImgCaption` to `true`

jyoung-uf committed:
Replace *all* instances of newline chars with line break element...

...not just the first.

jyoung-uf committed:
Use grid layout for thumbnail view wrapper

The current float-based layout for image thumbnails results in
irregular gaps due to the variability in image caption height.

jyoung-uf committed:
Bump line height for improved readability

jyoung-uf reviewed (commented):
Here's an example of the layout issue we observed due to longer caption lengths that affect rendering height. The existing css float rule remains in place to support older browsers, if needed.

<img width="1280" alt="float-layout-issue" src="https://github.com/specify/webportal-installer/assets/52349456/8910e942-3dbe-4679-b2af-6dfc0007a976">

---
Commit w/ potential fix: d628eace8807200f003c14b9d842a9634114dd30
